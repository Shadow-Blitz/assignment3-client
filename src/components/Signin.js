import React, { Component} from 'react'
import logo from '../assets/logo.svg'
import { Paper, Button, TextField } from '@material-ui/core'

const style = {
    Paper: {
        'width': '500px',
        'margin': '10% auto 0 auto',
        'textAlign': 'center',
        'padding': '5%'

    }
}

class Signin extends Component {

    render() {
        return (
            <Paper style={style.Paper}>
                <img src={logo}></img>
                <form>
                    {/* <input type='email' id='email' name="email" placeholder='Enter your email' required /><br /> */}
                    {/* <input type='password' id='password' name="password" placeholder='Password' required /><br /> */}

                    <TextField
                        required
                        id="standard-required"
                        label="Email"
                        margin="normal"
                        type="email"
                    />
                    <br />
                    <TextField
                        required
                        id="standard-required"
                        label="Password"
                        margin="normal"
                        type="password"
                    />
                    <br />
                    <Button type="submit" variant="contained" color="primary">Sign In</Button>
                    <Button type="button" variant='contained' color="primary" style={{ 'backgroundColor': 'orange' }}>Register</Button>
                </form>

            </Paper>
        )
    }
}

export default Signin

// export default ({ handleSignIn, loginError }) => (
//     <form onSubmit={handleSignIn}>
//         {loginError && <p>{loginError}</p>}
//         <label>Email: <input type="email" name="email" /></label><br />
//         <label>Password: <input type="password" name="password" /></label><br />
//         <button type="submit">Login</button>
//     </form>
// )