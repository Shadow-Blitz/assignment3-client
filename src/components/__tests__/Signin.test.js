import React from 'react';
import Signin from '../Signin';
import { shallow } from 'enzyme';

describe("Signin", () => {
    it("should render correctly", () => {
        const component = shallow(<Signin />);
    });
    it("should render initial layout", () => {
        // when
        const component = shallow(<Signin />);
        // then
        expect(component.getElements()).toMatchSnapshot();
    });
})