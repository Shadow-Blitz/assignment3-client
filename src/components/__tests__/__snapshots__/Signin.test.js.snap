// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin should render initial layout 1`] = `
Array [
  <WithStyles(Paper)
    style={
      Object {
        "margin": "10% auto 0 auto",
        "padding": "5%",
        "textAlign": "center",
        "width": "500px",
      }
    }
  >
    <img
      alt="Girl Guides Australia Logo"
      src="logo.svg"
    />
    <form>
      <TextField
        id="email"
        label="Email"
        margin="normal"
        required={true}
        select={false}
        type="email"
        variant="standard"
      />
      <br />
      <TextField
        id="password"
        label="Password"
        margin="normal"
        required={true}
        select={false}
        type="password"
        variant="standard"
      />
      <br />
      <WithStyles(Button)
        color="primary"
        style={
          Object {
            "margin": 15,
          }
        }
        type="submit"
        variant="contained"
      >
        Sign In
      </WithStyles(Button)>
      <Link
        replace={false}
        to="/signup"
      >
        <WithStyles(Button)
          color="primary"
          style={
            Object {
              "backgroundColor": "orange",
              "margin": 15,
            }
          }
          type="button"
          variant="contained"
        >
          Sign Up
        </WithStyles(Button)>
      </Link>
    </form>
  </WithStyles(Paper)>,
]
`;
